presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -163,18 +163,28 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
     int getPositionCount();
 
     /**
-     * Returns the logical size of this block in memory.
+     * Returns the size of this block as if it was compacted, ignoring any over-allocations.
```

Comment:
Let's add a second sentence to give a bit more clarity:
```
For example, in dictionary blocks, this only counts each dictionary entry once,
rather than each time a value is referenced.

Reply:
Will update the comment based on  your suggestion.
Apart from that do you think the java doc should also reflect that for blocks returned by `Block.getRegion()` this will give the size of the 'view' instead of the size of the underlying block? I suspected it's already adequately captured by '_size of this block as if it was compacted_'.